spec/mailers/user_mailer_spec.rb

```diff
@@ -1,6 +1,114 @@
 require "spec_helper"
 
 describe UserMailer do
+  describe "creatorship invitation" do
+    title = Faker::Book.title
+
+    subject(:email) { UserMailer.creatorship_invitation(work_creatorship.id, author.id).deliver }
+
+    let(:author) { create(:user) }
+    let(:second_author) { create(:user) }
+    let(:work) { create(:work, title: title, authors: [author.default_pseud, second_author.default_pseud]) }
+    let(:work_creatorship) { Creatorship.find_by(creation_id: work.id, pseud_id: second_author.default_pseud.id) }
+
+    # Test the headers
+    it_behaves_like "an email with a valid sender"
+
+    it "has the correct subject line" do
+      subject = "[#{ArchiveConfig.APP_SHORT_NAME}] Co-creator invitation"
+      expect(email).to have_subject(subject)
+    end
+
+    # Test both body contents
+    it_behaves_like "a multipart email"
+
+    it_behaves_like "a translated email"
+
+    describe "HTML version" do
+      it "has the correct content" do
+        expect(email).to have_html_part_content("to be listed as a co-creator")
+      end
+    end
+
+    describe "text version" do
+      it "has the correct content" do
+        expect(email).to have_text_part_content("to be listed as a co-creator")
+      end
+    end
+  end
+
+  describe "creatorship_notification" do
+    title = Faker::Book.title
+
+    subject(:email) { UserMailer.creatorship_notification(work_creatorship.id, author.id).deliver }
+
+    let(:author) { create(:user) }
+    let(:second_author) { create(:user) }
+    let(:work) { create(:work, title: title, authors: [author.default_pseud, second_author.default_pseud]) }
+    let(:work_creatorship) { Creatorship.find_by(creation_id: work.id, pseud_id: second_author.default_pseud.id) }
+
+    # Test the headers
+    it_behaves_like "an email with a valid sender"
+
+    it "has the correct subject line" do
+      subject = "[#{ArchiveConfig.APP_SHORT_NAME}] Co-creator notification"
+      expect(email).to have_subject(subject)
+    end
+
+    # Test both body contents
+    it_behaves_like "a multipart email"
+
+    it_behaves_like "a translated email"
+
+    describe "HTML version" do
```

Comment:
Are there any variations in the HTML and text content that we could check for? At a quick glance, it looks like the text version says `You can accept or reject this invitation on your Creator Invitations page:` while the HTML version says `You can accept or reject this invitation on your Creator Invitations page.`, but there might be something easier/better.

Reply:
Yeah, will add something to all of them